### ISSUE TITLE
Automated runs via Papermill

### ISSUE THREAD

**Issue by dylanwilder** [jupyter-incubator/sparkmagic#651]
**Is your feature request related to a problem? Please describe.**
I'm trying to determine whether notebooks using livy/sparkmagic can be run via papermill. It would seem that the answer is yes (papermill is mentioned in your docs), but I see no examples and am unclear how spark sessions can be bootstrapped non-interactively.  Automated session creation in general would be a great feature to add/document.

**Describe the solution you'd like**
An example which shows how to create spark sessions automatically.

**Comment by devstein** [comment]
Hi @dylanwilder thanks for making an issue!

In a (Py)Spark kernel, `sparkmagic` manages spark sessions for you, so there is no difference when running a notebook interactively or executing it via `papermill`. 

The only setup required is making the Sparkmagic (Spark, PySpark, etc) kernels available to `papermill`. This is typically done by forking `papermill`, feel free checkout and use [my fork](https://github.com/viaduct-ai/papermill).

Let me know if this helps!

**Comment by juhoautio** [comment]
I may be missing the point, but I have used regular Python3 kernel with spark magic magics in the notebook. Of course each cell that's supposed to run in the spark session needs to have the `%%spark` header then. But then I've executed such notebooks with papermill – nothing special needed. I would recommend [setting cleanup_all_sessions_on_exit=True in spark magic conf](https://github.com/jupyter-incubator/sparkmagic#papermill) though.
